src/RegisterTypeAbstractor.cpp

```diff
@@ -1,6 +1,125 @@
+    // the reading is not being synchronous anyway, we go for raw data as this gives a visible benefit for the user.
+    if(not registerInfo.isWriteable()) {
+      try {
+        dummyWritableAccessor.replace(device.getTwoDRegisterAccessor<decltype(arg)>(
+            registerInfo.getRegisterName() / ".DUMMY_WRITEABLE", nElements, 0, {ChimeraTK::AccessMode::raw}));
+      }
+      catch(...) {
+      }
+    }
+
+    accessor.replace(device.getTwoDRegisterAccessor<decltype(arg)>(
+        registerInfo.getRegisterName(), nElements, 0, {ChimeraTK::AccessMode::raw}));
+    returnValue = std::make_shared<RegisterTypeAbstractorRawImpl<decltype(arg), USER_DATA_TYPE>>(
+        accessor, dataDescriptor.rawDataType());
+    if(dummyWritableAccessor.isInitialised()) {
+      dummyReturnValue = std::make_shared<RegisterTypeAbstractorRawImpl<decltype(arg), USER_DATA_TYPE>>(
+          dummyWritableAccessor, dataDescriptor.rawDataType());
+    }
+  };
+  callForTypeNoVoid(dataDescriptor.rawDataType(), rawAccessorCreatorLambda);
+  return std::pair(returnValue, dummyReturnValue);
+}
+
+std::pair<std::shared_ptr<RegisterTypeAbstractor>, std::shared_ptr<RegisterTypeAbstractor>> createAbstractAccessors(
+    ChimeraTK::RegisterInfo const& registerInfo, ChimeraTK::Device& device) {
+  auto dataDescriptor = registerInfo.getDataDescriptor();
+  switch(dataDescriptor.fundamentalType()) {
+    case ChimeraTK::DataDescriptor::FundamentalType::numeric:
+      if(dataDescriptor.isIntegral()) {
+        if(dataDescriptor.isSigned()) {
+          return createTypedAccessors<int32_t>(registerInfo, device);
+        }
+        return createTypedAccessors<uint32_t>(registerInfo, device);
+      }
+      else { // not integral, use double
+        return createTypedAccessors<double>(registerInfo, device);
+      }
+      break;
+    case ChimeraTK::DataDescriptor::FundamentalType::string:
+      return createTypedAccessors<std::string>(registerInfo, device);
+      break;
+    case ChimeraTK::DataDescriptor::FundamentalType::boolean:
+      return createTypedAccessors<uint32_t>(registerInfo, device);
+      break;
+    case ChimeraTK::DataDescriptor::FundamentalType::nodata:
+      // The accessor might be writeable, and even if there is not data the update time is displayed.
+      // Create an int accessor (it's always a 2d accessor which cannot be void).
+      return createTypedAccessors<int32_t>(registerInfo, device);
+      break;
+    case ChimeraTK::DataDescriptor::FundamentalType::undefined:
+      // fall into default, just mentioned for completeness
+    default:
+      return std::make_pair(nullptr, nullptr);
+      ;
+  }
+}
+
+//======================================================================================================
+//======================================================================================================
+
 template<class USER_DATA_TYPE>
 std::shared_ptr<RegisterTypeAbstractor> createTypedAccessor(
```

Comment:
Remove the old functions. I think they are not needed any more.

Reply:
Yes, completely forgot about it.